### PR TITLE
Appearance Changes

### PR DESCRIPTION
**Pull request by mattregul**
- Now uses weblegends style.css
- Has own stylesheet (aistyle.css) for df-ai specific items (nav menu & stock report's table)
- Chopped original nav menu up into chunks (likely over-engineered)
- Added “Home” link, to return to weblegends homepage

image

### CODE REVIEW

File: weblegends.cpp [hunk 1]
```diff
@@ -14,6 +14,16 @@ extern bool & enabled;
 
 REQUIRE_GLOBAL(world);
 
+enum AIPages {
+    Status,             // Status
+    //Report,           // Report - No longer used
```
Comment (BenLubar) [inline weblegends.cpp]: We don't need the `Report` enum item, even in a comment. It just makes the code messier.

File: weblegends.cpp [hunk 2]
```diff
@@ -35,9 +45,124 @@ std::string html_escape(const std::string & str)
...
+            title = "DF-AI - Blueprints";
+            break;
+        case AIPages::Version:
+            title = "DF-AI - Version";
+            break;
+        default: // default to status page
+            title = "DF-AI - Status";
+    }
+    // Build Nav Menu =========
+    handler.cp437_out() << "<!DOCTYPE html><html lang=\"en\"><head><title>" << title << "</title><base href=\"../..\"/><link rel=\"stylesheet\" href=\"style.css\"/><link rel=\"stylesheet\" href=\"df-ai/aistyle.css\"/></head>";
```
Comment (BenLubar) [inline weblegends.cpp]: `<base href="../.."/>` should only be used for pages with two slashes in their relative url. `<base href=".."/>` is for pages like `df-ai/version` and `df-ai/plan`, and there should not be a `base` tag at all for the `df-ai` status page.

File: weblegends.cpp [hunk 3]
```diff
@@ -35,9 +45,124 @@ std::string html_escape(const std::string & str)
...
+    // close P from beginning of menu
+    handler.cp437_out() << "</p>";
+    return;
+}
+
+
 bool ai_weblegends_handler(weblegends_handler_v1 & handler, const std::string & url)
 {
-    if (url == "/style.css")
+    // Requesting Sytlesheet
```
Comment (BenLubar) [inline weblegends.cpp]: You spelled Stylesheet wrong here 😛

File: weblegends.cpp [hunk 4]
```diff
@@ -59,27 +184,26 @@ bool ai_weblegends_handler(weblegends_handler_v1 & handler, const std::string &
...
         return true;
     }
 
+    // df-ai is not loaded or enabled
     if (!enabled || !dwarfAI)
     {
         handler.status_code() = 503;
         handler.status_description() = "Service Unavailable";
-        handler.cp437_out() << "<!DOCTYPE html><html lang=\"en\"><head><title>df-ai</title></head>";
+        handler.cp437_out() << "<!DOCTYPE html><html lang=\"en\"><head><title>df-ai - Service Unavailable</title></head>";
```
Comment (BenLubar) [inline weblegends.cpp]: `Service Unavailable` is probably too technical for most df-ai users (the previous line is part of the HTTP implementation.) Maybe something like `df-ai - not active`?

Also, this page does not include any stylesheets. The `base` tag might be a bit complicated to implement here.

**Comment by mattregul** [comment]
I *think* I did the rebase right?

**Comment by mattregul** [comment]
`The "Not Active" screen`
image

`Latest Render`
image

File: weblegends.cpp [hunk 5]
```diff
@@ -45,114 +46,115 @@ std::string html_escape(const std::string & str)
...
+            baseURL = "../";
             break;
         case AIPages::Version:
             title = "DF-AI - Version";
+            baseURL = "../";
             break;
-        default: // default to status page
+        default: // default to status page, unless DF-AI isn't active
             title = "DF-AI - Status";
+            baseURL = "/";
```
Comment (BenLubar) [inline weblegends.cpp]: This should be `./` or just an empty string. Otherwise, it's like an infinite number of `../../..`.

File: weblegends.cpp [hunk 6]
```diff
@@ -185,21 +187,35 @@ bool ai_weblegends_handler(weblegends_handler_v1 & handler, const std::string &
...
+    create_nav_menu(handler,url);
+
     // df-ai is not loaded or enabled
     if (!enabled || !dwarfAI)
     {
         handler.status_code() = 503;
-        handler.status_description() = "Service Unavailable";
-        handler.cp437_out() << "<!DOCTYPE html><html lang=\"en\"><head><title>df-ai - Service Unavailable</title></head>";
-        handler.cp437_out() << "<body><p><i>AI is not active.</i></p></body></html>";
+        handler.status_description() = "df-ai - Not Active";
```
Comment (BenLubar) [inline weblegends.cpp]: That should be `Service Unavailable`. `status_description` isn't shown to people, it's for the benefit of machines.

**Pull request merged**